### PR TITLE
Deprecate Raise.With<TEventHandler>

### LINKED ISSUE #1153: Deprecate Raise.With<TEventHandler>
In preparation for #1152.

### PR DESCRIPTION
**Pull request by blairconrad**
Fixes #1153.

### CODE REVIEW

File: docs/raising-events.md [hunk 1]
```diff
@@ -58,7 +58,8 @@ fake.CustomEvent += Raise.FreeForm.With(fake, sampleCustomEventArgs);
 fake.FreeformEvent += Raise.FreeForm.With(7);
 ```
 
-* or specify the delegate type explicitly as a type parameter to `Raise.With`:
+* or specify the delegate type explicitly as a type parameter to `Raise.With`
+  **(note that this syntax has been deprecated and will be removed in FakeItEasy 5.0.0)**:
```
Comment (thomaslevesque) [inline docs/raising-events.md]: If we deprecate it, is it still worth mentioning in the docs?

Comment (blairconrad) [inline docs/raising-events.md]: You're a smart man.

Comment (blairconrad) [inline docs/raising-events.md]: amended

**Comment by blairconrad** [comment]
Closing in favour of #1155.

**Pull request closed without merging**